### PR TITLE
Migrated from docker attach to fluentd log-driver

### PR DESCRIPTION
**Pull request by Luzifer**
In order to not stream all logs through the `docker.sock` and to fetch all logs of a container, even those produced in the first second of the container creation switch from the `docker attach` approach to providing an endpoint suitable for receiving logs through the fluentd log-driver.

### CODE REVIEW

**Comment by Luzifer** [comment]
@Jimdo/Werkzeugschmiede please take a look at this!

**Comment by mlafeldt** [comment]
Could you add some more info like the in-order-to and what this does on a higher level?

File: main.go [hunk 1]
```diff
@@ -70,86 +73,84 @@ func main() {
...
+	return nil
 }
 
-func inSlice(s []string, k string) bool {
-	for _, v := range s {
-		if v == k {
-			return true
-		}
+func getContainerInformation(id string) (*docker.Container, error) {
+	if c, ok := containerCache[id]; ok {
```
Comment (mlafeldt) [inline main.go:164]: We need to guard this map with a lock.

File: main.go [hunk 2]
```diff
@@ -70,86 +73,84 @@ func main() {
...
+		go handleFluentdForwardConnection(conn)
+	}
+}
+
+func handleFluentdForwardConnection(c net.Conn) {
+	defer c.Close()
+	buffer := bytes.NewBuffer([]byte{})
+
+	for {
+		tmp := make([]byte, 256)
```
Comment (mlafeldt) [inline main.go]: Not sure if this has any performance impact, but this could be allocated once per connection. Maybe even on the stack?

**Comment by mlafeldt** [comment]
LGTM

**Comment by mlafeldt** [comment]
skateboard

**Pull request merged**